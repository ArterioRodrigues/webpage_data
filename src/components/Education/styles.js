/**
 * Education component styles.
 */
import { mSize, orange, cyan, textDark } from '../../theme';

export default () => ({
  entry: {
    marginBottom: '30px',
    '&>div': {
      marginBottom: '4px',
    },
  },
  school: {
    fontSize: mSize,
  },
  department: {
    color: orange,
    fontWeight: '500',
  },
  degree: {
    color: cyan,
    fontWeight: '500',
  },
  years: {
    color: textDark,
    fontStyle: 'italic',
  },
});
