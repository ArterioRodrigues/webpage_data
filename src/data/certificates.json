{
  "sectionTitle": "certifications",
  "sectionSubtitle": "Certificates that I have collected over time",
  "content": [
    {
      "name": "Responsive Web Design Certification",
      "url": "https://www.freecodecamp.org/certification/caglarturali/responsive-web-design"
    },
    {
      "name": "Introduction to Game Development",
      "url": "https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M"
    },
    {
      "name": "Introduction to Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG"
    },
    {
      "name": "3D Models for Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV"
    },
    {
      "name": "The Art of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF"
    },
    {
      "name": "The Technology of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER"
    },
    {
      "name": "Pro Tools Basics",
      "url": "https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F"
    }
  ]
}
