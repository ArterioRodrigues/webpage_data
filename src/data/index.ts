export * from './certificates';
export * from './contact';
export * from './education';
export * from './greeter';
export * from './metadata';
export * from './projects';
export * from './skills';
