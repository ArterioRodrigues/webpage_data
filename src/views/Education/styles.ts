/**
 * Education component styles.
 */
import { Styles } from 'react-jss';
import { mSize, orange, cyan, textDark } from '../../theme';

export default {
  entry: {
    marginBottom: '30px',
    '&>div': {
      marginBottom: '4px',
    },
  },
  school: {
    fontSize: mSize,
  },
  department: {
    color: orange,
    fontWeight: 500,
  },
  degree: {
    color: cyan,
    fontWeight: 500,
  },
  years: {
    color: textDark,
    fontStyle: 'italic',
  },
} as Styles;
