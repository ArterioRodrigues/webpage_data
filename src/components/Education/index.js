/**
 * Education component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../Container';
import education from '../../data/education.json';
import styles from './styles';

const useStyles = createUseStyles(styles);

const Education = () => {
  const classes = useStyles();

  return (
    <Container
      title={education.sectionTitle}
      subtitle={education.sectionSubtitle}
    >
      <ul>
        {education.content.map((entry) => (
          <div className="container__list" key={entry.years}>
            <li key={entry.years} className="education__entry">
              <div className={classes.school}>{entry.school}</div>
              <div className={classes.department}>{entry.department}</div>
              <div className={classes.degree}>{entry.degree}</div>
              <div className={classes.years}>{entry.years}</div>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Education;
