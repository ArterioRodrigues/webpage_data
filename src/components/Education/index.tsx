/**
 * Education component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../Container';
import { Page, School } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface EducationProps {
  data: Page<School>;
}

const Education: React.FC<EducationProps> = ({ data }) => {
  const classes = useStyles();

  return (
    <Container title={data.name} subtitle={data.desc}>
      <ul>
        {data.content.map(({ name, department, degree, years }, i) => (
          <div key={`education-item-${i}`}>
            <li className={classes.entry}>
              <div className={classes.school}>{name}</div>
              <div className={classes.department}>{department}</div>
              <div className={classes.degree}>{degree}</div>
              <div
                className={classes.years}
              >{`${years.start} - ${years.end}`}</div>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Education;
