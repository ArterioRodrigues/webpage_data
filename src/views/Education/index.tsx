/**
 * Education component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../../components/Container';
import { School, BaseProps } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export type EducationProps = BaseProps<School>;

const Education: React.FC<EducationProps> = ({ data, fileName }) => {
  const classes = useStyles();

  return (
    <Container
      title={data.name}
      subtitle={data.desc}
      seo={{ title: fileName, description: data.desc }}
    >
      <ul>
        {data.content.map(({ name, department, degree, years }, i) => (
          <div key={`education-item-${i}`}>
            <li className={classes.entry}>
              <div className={classes.school}>{name}</div>
              <div className={classes.department}>{department}</div>
              <div className={classes.degree}>{degree}</div>
              <div
                className={classes.years}
              >{`${years.start} - ${years.end}`}</div>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Education;
