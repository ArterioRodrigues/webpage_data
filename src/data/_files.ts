import { NavLink } from '../models';
import Greeter from '../views/Greeter';
import Skills from '../views/Skills';
import Education from '../views/Education';
import Projects from '../views/Projects';
import Certificates from '../views/Certificates';
import Contact from '../views/Contact';
import { certificates, contact, education, projects, skills } from '.';

export const files: NavLink[] = [
  {
    icon: ['fab', 'html5'],
    color: '#e44d26',
    url: '/',
    name: 'index.html',
    route: {
      component: Greeter,
      data: contact,
    },
  },
  {
    icon: ['fab', 'js'],
    color: '#ffca28',
    url: '/skills',
    name: 'skills.js',
    route: {
      component: Skills,
      data: skills,
    },
  },
  {
    icon: 'university',
    color: '#42a5f5',
    url: '/education',
    name: '.educationrc',
    route: {
      component: Education,
      data: education,
    },
  },
  {
    icon: 'sliders-h',
    color: '#98c379',
    url: '/projects',
    name: 'projects.config',
    route: {
      component: Projects,
      data: projects,
    },
  },
  {
    icon: ['fab', 'sass'],
    color: '#e13e76',
    url: '/certifications',
    name: 'certifications.sass',
    route: {
      component: Certificates,
      data: certificates,
    },
  },
  {
    icon: ['fab', 'markdown'],
    color: '#abb2bf',
    url: '/contact',
    name: 'contact.md',
    route: {
      component: Contact,
      data: contact,
    },
  },
];
