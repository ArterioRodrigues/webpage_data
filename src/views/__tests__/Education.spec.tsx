import React from 'react';
import { render } from '@testing-library/react';
import Education from '../Education';
import { education } from '../../data';

describe('<Education />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <Education data={education} fileName="education.test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
