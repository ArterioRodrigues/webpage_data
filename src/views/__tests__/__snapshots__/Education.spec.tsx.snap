// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Education /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-6"
  >
    <h2
      class="title-0-2-7"
    >
      education
    </h2>
    <h3
      class="subtitle-0-2-8"
    >
      My academic background
    </h3>
    <div
      class="content-0-2-9"
    >
      <ul>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="school-0-2-2"
            >
              Kirikkale University
            </div>
            <div
              class="department-0-2-3"
            >
              Computer Education and Instructional Technologies
            </div>
            <div
              class="degree-0-2-4"
            >
              Bachelor's Degree
            </div>
            <div
              class="years-0-2-5"
            >
              2010 - 2017
            </div>
          </li>
        </div>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="school-0-2-2"
            >
              Kirikkale University
            </div>
            <div
              class="department-0-2-3"
            >
              Physics
            </div>
            <div
              class="degree-0-2-4"
            >
              [Dropout]
            </div>
            <div
              class="years-0-2-5"
            >
              2008 - 2010
            </div>
          </li>
        </div>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
